main/test/src/util/ScriptTestSuite.scala

```diff
@@ -52,20 +52,43 @@ abstract class ScriptTestSuite(fork: Boolean) extends TestSuite {
     wd = wd,
     initialSystemProperties = sys.props.toMap
   )
+  lazy val runner = runnerStdout(System.out)
   def eval(s: String*): Boolean = {
-    if (!fork) runner.runScript(workspacePath / buildPath, s.toList)
-    else {
-      try {
-        os.proc(os.home / "mill-release", "-i", s).call(
-          wd,
-          stdin = os.Inherit,
-          stdout = os.Inherit,
-          stderr = os.Inherit
-        )
-        true
-      } catch { case e: Throwable => false }
+    if (!fork) {
+      runner.runScript(workspacePath / buildPath, s.toList)
+    } else {
+      evalFork(os.Inherit, s)
     }
   }
+  def evalStdout(s: String*): (Boolean, Seq[String]) = {
+    if (!fork) {
+      val outputStream = new ByteArrayOutputStream
+      val printStream = new PrintStream(outputStream)
+      val result = runnerStdout(printStream).runScript(workspacePath / buildPath, s.toList)
+      val stdoutArray = outputStream.toByteArray
+      val stdout: Seq[String] =
+        if (stdoutArray.isEmpty) Seq.empty
+        else new String(stdoutArray).split('\n')
+      (result, stdout)
+    } else {
+      val output = Seq.newBuilder[String]
+      val processOutput = os.ProcessOutput.Readlines(output += _)
+
+      val result = evalFork(processOutput, s)
+      (result, output.result())
+    }
+  }
+  private def evalFork(stdout: os.ProcessOutput, s: Seq[String]): Boolean = {
+    try {
+      os.proc(os.home / "mill-release", "-i", s).call(
```

Comment:
Unrelated to your PR, but we should get rid of this hardcoded `$HOME/mill-release` thing.